ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp

```diff
@@ -79,9 +79,52 @@ dwarf_init(int fd,
 	 */
 	FILE *fp = NULL;
 	if (DW_DLV_OK == ret) {
-		stringstream ss;
-		ss << "dwarfdump " << filepath << " 2>&1";
-		fp = popen(ss.str().c_str(), "r");
+
+		/* On some Mac machines, calling "dwarfdump" in shell calls a version named
+		 * "llvm-dwarfdump" instead of the version of dwarfdump DDR uses, which is
+		 * instead named "dwarfdump-classic" on those systems. Thus, we first try to
+		 * find "dwarfdump-classic" directly in case "dwarfdump" is linked to
+		 * "llvm-dwarfdump". If we don't find "dwarfdump-classic", then the machine
+		 * we are on uses the correct version of dwarfdump.
+		 */
+		fp = popen("xcrun -f dwarfdump-classic 2>/dev/null", "r");
+		if (NULL == fp) {
+			fp = popen("xcrun -f dwarfdump 2>/dev/null", "r");
```

Comment:
I don't like that this `popen` call is duplicated below. Perhaps we need a helper function that does the `popen`, `getline` and path check, so the flow here could be something like this:
```
if (toolCheck(&toolpath, "xcrun -f dwarfdump-classic 2>/dev/null")
||  toolCheck(&toolpath, "xcrun -f dwarfdump 2>/dev/null")
) {
    // use 'toolpath'
}
```